### PR TITLE
CI/Test with GitHub Actions

### PR DESCRIPTION
**Pull request by johnsyweb**
#### Context

With wait times getting longer on our previous CI provider, we have decided to use GitHub Actions for our open-source projects.

#### Change

 - Defined CI/Test action in .github/workflows/ci.yml and updated badge.
 - Updated CHANGELOG
 - Use HEAD commit in URIs to make it easier to change default branch name

### CODE REVIEW

**Review by orien — APPROVED** [review]
+1

**Review by lucascaton — APPROVED** [review]
Looks good to me!

File: README.md [hunk 1]
```diff
@@ -137,16 +137,16 @@ attacker. For this reason we advise you not to store the results of
  - [Matthieu Aussaguel](https://github.com/matthieua)
  - [_et al._](https://github.com/envato/zxcvbn-ruby/graphs/contributors)
 
-## License [![license](https://img.shields.io/github/license/mashape/apistatus.svg?style=flat-square)](https://github.com/envato/zxcvbn-ruby/blob/master/LICENSE.txt)
+## License [![license](https://img.shields.io/github/license/mashape/apistatus.svg?style=flat-square)](https://github.com/envato/zxcvbn-ruby/blob/HEAD/LICENSE.txt)
 
 `zxcvbn-ruby` uses MIT license, the same as [zxcvbn.js][zxcvbn.js] itself. See
-[`LICENSE.txt`](https://github.com/envato/zxcvbn-ruby/blob/master/LICENSE.txt)
+[`LICENSE.txt`](https://github.com/envato/zxcvbn-ruby/blob/HEAD/LICENSE.txt)
```
Comment (johnsyweb) [inline README.md:143]: @stevehodgkiss: I'd like to rename the default branch to `main` but I don't seem to have permissions to do so on this repository. Is this something you can do, please?

Comment (stevehodgkiss) [reply README.md:143]: @johnsyweb sorry I can't help - I don't have admin on the repo

**Pull request merged**